src/scancode/cli.py

```diff
@@ -572,11 +583,25 @@ def scan_one(input_file, scanners, diag=False):
     return scan_result
 
 
-def save_results(files_count, scanned_files, format, input, output_file):
+def has_findings(active_scans, file_data):
+    for scan_name in active_scans:
+        if file_data.get(scan_name):
+            return True
+
+    return False
+
+
+def save_results(scanners, only_findings, files_count, scanned_files, format, input, output_file):
     """
     Save scan results to file or screen.
     """
-    # note: in tests, sys.sdtout is not used, but some io wrapper with no name attributes
+
+    if only_findings:
+        active_scans = [k for k, v in scanners.items() if v[0] and v[1]]
+        scanned_files = [file_data for file_data in scanned_files if has_findings(active_scans, file_data)]
+        files_count = len(scanned_files)
```

Comment:
It's probably better to not add files without findings in the first place (if that option is given on the command line) rather than filtering the results afterwards...